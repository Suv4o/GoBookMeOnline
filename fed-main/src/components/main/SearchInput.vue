<script lang="ts">
export default {
  name: 'SearchInput',
}
</script>

<script setup lang="ts">
import { computed, ref } from 'vue'
import { SearchIcon } from '@heroicons/vue/solid'
import { ChevronRightIcon, UsersIcon } from '@heroicons/vue/outline'
import { Combobox, ComboboxInput, ComboboxOptions, ComboboxOption } from '@headlessui/vue'

const people = [
  {
    id: 1,
    name: 'Leslie Alexander',
    phone: '1-493-747-9031',
    email: 'lesliealexander@example.com',
    role: 'Co-Founder / CEO',
    url: 'https://example.com',
    profileUrl: '#',
    imageUrl:
      'https://images.unsplash.com/photo-1494790108377-be9c29b29330?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=facearea&facepad=2&w=256&h=256&q=80',
  },
  {
    id: 2,
    name: 'Juniper Moody',
    phone: '1-582-300-8610',
    email: 'lesliealexander@example.com',
    role: 'Co-Founder / CEO',
    url: 'https://example.com',
    profileUrl: '#',
    imageUrl:
      'https://images.unsplash.com/photo-1494790108377-be9c29b29330?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=facearea&facepad=2&w=256&h=256&q=80',
  },
  {
    id: 3,
    name: 'Miyah Brady',
    phone: '1-316-938-5949',
    email: 'lesliealexander@example.com',
    role: 'Co-Founder / CEO',
    url: 'https://example.com',
    profileUrl: '#',
    imageUrl:
      'https://images.unsplash.com/photo-1494790108377-be9c29b29330?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=facearea&facepad=2&w=256&h=256&q=80',
  },
  {
    id: 4,
    name: 'Ellen Walker',
    phone: '1-203-368-8795',
    email: 'lesliealexander@example.com',
    role: 'Co-Founder / CEO',
    url: 'https://example.com',
    profileUrl: '#',
    imageUrl:
      'https://images.unsplash.com/photo-1494790108377-be9c29b29330?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=facearea&facepad=2&w=256&h=256&q=80',
  },
  {
    id: 5,
    name: 'Emma Nicholson',
    phone: '1-248-393-7011',
    email: 'lesliealexander@example.com',
    role: 'Co-Founder / CEO',
    url: 'https://example.com',
    profileUrl: '#',
    imageUrl:
      'https://images.unsplash.com/photo-1494790108377-be9c29b29330?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=facearea&facepad=2&w=256&h=256&q=80',
  },
]

// const recent = [people[0], people[1], people[2], people[3], people[4]]
const recent: any = []
const isFocused = ref(false)

const query = ref('')
const filteredPeople = computed(() =>
  query.value === ''
    ? []
    : people.filter(person => {
        return person.name.toLowerCase().includes(query.value.toLowerCase())
      })
)

function onSelect(person: any) {
  window.location = person.url
}

function setRecentSearch() {
  isFocused.value = true
}

function unsetRecentSearch() {
  isFocused.value = false
}
</script>

<template>
  <Combobox
    v-slot="{ activeOption }"
    as="div"
    class="transform divide-y divide-gray-100 overflow-hidden rounded-xl bg-white shadow-2xl ring-1 ring-black ring-opacity-5 transition-all"
    @update:model-value="onSelect"
  >
    <div class="relative">
      <SearchIcon class="pointer-events-none absolute top-3.5 left-4 h-5 w-5 text-gray-400" aria-hidden="true" />
      <ComboboxInput
        class="h-12 w-full border-0 bg-transparent pl-11 pr-4 text-gray-800 placeholder-gray-400 focus:ring-0 sm:text-sm"
        placeholder="Search..."
        @focus="setRecentSearch"
        @blur="unsetRecentSearch"
        @change="query = $event.target.value"
      />
    </div>

    <ComboboxOptions
      v-if="query === '' || filteredPeople.length > 0"
      class="flex divide-x divide-gray-100"
      as="div"
      static
      hold
    >
      <div
        v-if="isFocused && (query !== '' || recent.length)"
        :class="['max-h-96 min-w-0 flex-auto scroll-py-4 overflow-y-auto px-6 py-4', activeOption && 'sm:h-96']"
      >
        <h2 v-if="query === '' && recent.length" class="mt-2 mb-4 text-xs font-semibold text-gray-500">
          Recent searches
        </h2>
        <div hold class="-mx-2 text-sm text-gray-700">
          <ComboboxOption
            v-for="person in query === '' ? recent : filteredPeople"
            :key="person.id"
            v-slot="{ active }"
            :value="person"
            as="template"
          >
            <div
              :class="[
                'group flex cursor-default select-none items-center rounded-md p-2',
                active && 'bg-gray-100 text-gray-900',
              ]"
            >
              <img :src="person.imageUrl" alt="" class="h-6 w-6 flex-none rounded-full" />
              <span class="ml-3 flex-auto truncate">{{ person.name }}</span>
              <ChevronRightIcon v-if="active" class="ml-3 h-5 w-5 flex-none text-gray-400" aria-hidden="true" />
            </div>
          </ComboboxOption>
        </div>
      </div>

      <div
        v-if="activeOption"
        class="hidden h-96 w-1/2 flex-none flex-col divide-y divide-gray-100 overflow-y-auto sm:flex"
      >
        <div class="flex-none p-6 text-center">
          <img :src="activeOption.imageUrl" alt="" class="mx-auto h-16 w-16 rounded-full" />
          <h2 class="mt-3 font-semibold text-gray-900">
            {{ activeOption.name }}
          </h2>
          <p class="text-sm leading-6 text-gray-500">{{ activeOption.role }}</p>
        </div>
        <div class="flex flex-auto flex-col justify-between p-6">
          <dl class="grid grid-cols-1 gap-x-6 gap-y-3 text-sm text-gray-700">
            <dt class="col-end-1 font-semibold text-gray-900">Phone</dt>
            <dd>{{ activeOption.phone }}</dd>
            <dt class="col-end-1 font-semibold text-gray-900">URL</dt>
            <dd class="truncate">
              <a :href="activeOption.url" class="text-teal-600 underline">
                {{ activeOption.url }}
              </a>
            </dd>
            <dt class="col-end-1 font-semibold text-gray-900">Email</dt>
            <dd class="truncate">
              <a :href="`mailto:${activeOption.email}`" class="text-teal-600 underline">
                {{ activeOption.email }}
              </a>
            </dd>
          </dl>
          <button
            type="button"
            class="mt-6 w-full rounded-md border border-transparent bg-teal-600 py-2 px-4 text-sm font-medium text-white shadow-sm hover:bg-teal-700 focus:outline-none focus:ring-2 focus:ring-teal-500 focus:ring-offset-2"
          >
            Send message
          </button>
        </div>
      </div>
    </ComboboxOptions>

    <div v-if="query !== '' && filteredPeople.length === 0" class="py-14 px-6 text-center text-sm sm:px-14">
      <UsersIcon class="mx-auto h-6 w-6 text-gray-400" aria-hidden="true" />
      <p class="mt-4 font-semibold text-gray-900">No people found</p>
      <p class="mt-2 text-gray-500">We couldnâ€™t find anything with that term. Please try again.</p>
    </div>
  </Combobox>
</template>
