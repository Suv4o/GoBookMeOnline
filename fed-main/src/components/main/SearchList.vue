<script lang="ts">
export default {
  name: 'SearchList',
}
</script>

<script setup lang="ts">
import BookSearchList from '../global/BookSearchList/BookSearchList.vue'
import BookPagination from '../global/BookPagination/BookPagination.vue'
</script>

<template>
  <book-search-list title="Locations" type="location" />
  <book-search-list title="People" type="person" />
  <div class="mt-4">
    <book-pagination />
  </div>
</template>
