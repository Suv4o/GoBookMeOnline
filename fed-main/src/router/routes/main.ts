import type { RouteRecordRaw } from 'vue-router'

const mainRoutes: Array<RouteRecordRaw> = [
  {
    path: '/',
    name: 'home',
    components: {
      default: () => import('../../views/HomePage.vue'),
      DefaultMainNav: () => import('../../components/Default/DefaultMainNav/DefaultMainNav.vue'),
      DefaultNewsletter: () => import('../../components/Default/DefaultNewsletter/DefaultNewsletter.vue'),
      DefaultFooter: () => import('../../components/Default/DefaultFooter/DefaultFooter.vue'),
    },
  },
  {
    path: '/:pathMatch(.*)*',
    name: 'notFound',
    components: {
      default: () => import('../../views/NotFoundPage.vue'),
    },
  },
]

export default mainRoutes
