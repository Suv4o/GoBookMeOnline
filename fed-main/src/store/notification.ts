import { defineStore } from 'pinia'

export enum NotificationTypes {
  Success = 'success',
  Error = 'error',
  Warning = 'warning',
  Info = 'info',
}

export interface NotificationState {
  isOpen: boolean
  type: NotificationTypes | null
  title: string | null
  message: string | null
}

export const useNotificationStore = defineStore('notificationStore', {
  state: (): NotificationState => {
    return {
      isOpen: false,
      type: null,
      title: null,
      message: null,
    }
  },
})
