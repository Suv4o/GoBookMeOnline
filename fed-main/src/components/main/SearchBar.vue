<script lang="ts">
export default {
  name: 'SearchBar',
}
</script>

<script setup lang="ts">
import SearchInput from './SearchInput.vue'
import LocationInput from './LocationInput.vue'
</script>

<template>
  <div class="flex flex-row mx-auto max-w-3xl">
    <div class="basis-2/3 mr-4">
      <search-input />
    </div>
    <div class="basis-1/3"><location-input /></div>
  </div>
</template>
