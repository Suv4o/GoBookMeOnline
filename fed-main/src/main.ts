import { createApp } from 'vue'
import App from './App.vue'
import router from './router'
import { createPinia } from 'pinia'
import './assets/tailwind/index.css'
import { auth } from './config/firebase.config'
import { onAuthStateChanged } from 'firebase/auth'
import { useAuthStore } from './store/auth'
export const pinia = createPinia()
const appConfig = { isAppMounted: false }

const app = createApp(App)
app.provide('$auth', auth)
app.use(pinia)
app.use(router)

onAuthStateChanged(auth, user => {
  if (!user) {
    const useStoreAuth = useAuthStore(pinia)
    useStoreAuth.accessToken = ''
    useStoreAuth.accessTokenExpirationTime = 0
    useStoreAuth.user = null
  }

  if (!appConfig.isAppMounted) {
    appConfig.isAppMounted = true
    router.isReady().then(() => {
      app.mount('#app')
    })
  }
})
