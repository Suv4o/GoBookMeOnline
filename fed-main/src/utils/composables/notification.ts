import { useNotificationStore } from '../../store/notification'
import { NotificationState } from '../../store/notification'
let notificationTimeout = null as ReturnType<typeof setTimeout> | null

export function useNotification(
  notificationOptions: Pick<NotificationState, 'type' | 'title' | 'message'>,
  timeout = 8000
) {
  const notificationStore = useNotificationStore()
  const { type, title, message } = notificationOptions

  if (notificationTimeout) {
    clearTimeout(notificationTimeout)
  }

  notificationStore.isOpen = true
  notificationStore.type = type
  notificationStore.title = title
  notificationStore.message = message

  notificationTimeout = setTimeout(() => {
    notificationStore.isOpen = false
    notificationStore.type = null
    notificationStore.title = null
    notificationStore.message = null
  }, timeout)
}
