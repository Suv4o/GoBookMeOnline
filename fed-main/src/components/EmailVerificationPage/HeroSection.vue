<script lang="ts">
export default {
  name: 'HeroSection',
}
</script>

<script setup lang="ts">
import { Assertions } from '../../types/guards'
import { useFetch } from '../../utils/composables/fetch'
import { parseErrorMessage } from '../../utils/helpers'
import { useNotification } from '../../utils/composables/notiofication'
import { NotificationTypes } from '../../store/notification'
import { ref } from 'vue'

const isProcessing = ref(false)

async function sendVerificationEmailLink() {
  try {
    isProcessing.value = true
    const { data, error } = await useFetch({
      url: '/user/email-verification',
      method: 'GET',
    })

    if (error.value) {
      throw new Error(parseErrorMessage(error.value.message))
    }

    useNotification({
      type: NotificationTypes.Success,
      title: 'Verification Email Sent',
      message: 'A new verification email has been sent to your email address.',
    })
    isProcessing.value = false
  } catch (error) {
    isProcessing.value = false
    Assertions.isError(error)
    useNotification({ type: NotificationTypes.Error, title: 'Error', message: error.message })
  }
}
</script>
<template>
  <div class="relative pt-28 pb-16 bg-white z-0">
    <div class="hidden absolute top-0 inset-x-0 h-1/2 bg-gray-50 lg:block" aria-hidden="true" />
    <div class="max-w-7xl mx-auto bg-teal-600 lg:bg-transparent lg:px-8">
      <div class="lg:grid lg:grid-cols-12">
        <div class="relative z-10 lg:col-start-1 lg:row-start-1 lg:col-span-4 lg:py-16 lg:bg-transparent">
          <div class="absolute inset-x-0 h-1/2 bg-gray-50 lg:hidden" aria-hidden="true" />
          <div class="max-w-md mx-auto px-4 sm:max-w-3xl sm:px-10 lg:max-w-none lg:p-0">
            <div class="aspect-w-10 aspect-h-6 sm:aspect-w-2 sm:aspect-h-1 lg:aspect-w-1">
              <img
                class="object-cover object-center rounded-3xl shadow-2xl"
                src="../../assets/images/hero/email-verification.jpg"
                alt=""
              />
            </div>
          </div>
        </div>

        <div
          class="relative bg-teal-600 lg:col-start-3 lg:row-start-1 lg:col-span-10 lg:rounded-3xl lg:grid lg:grid-cols-10 lg:items-center"
        >
          <div class="hidden absolute inset-0 overflow-hidden rounded-3xl lg:block" aria-hidden="true">
            <svg
              class="absolute bottom-full left-full transform translate-y-1/3 -translate-x-2/3 xl:bottom-auto xl:top-0 xl:translate-y-0"
              width="404"
              height="384"
              fill="none"
              viewBox="0 0 404 384"
              aria-hidden="true"
            >
              <defs>
                <pattern
                  id="64e643ad-2176-4f86-b3d7-f2c5da3b6a6d"
                  x="0"
                  y="0"
                  width="20"
                  height="20"
                  patternUnits="userSpaceOnUse"
                >
                  <rect x="0" y="0" width="4" height="4" class="text-teal-800" fill="currentColor" />
                </pattern>
              </defs>
              <rect width="404" height="384" fill="url(#64e643ad-2176-4f86-b3d7-f2c5da3b6a6d)" />
            </svg>
            <svg
              class="absolute top-full transform -translate-y-1/3 -translate-x-1/3 xl:-translate-y-1/2"
              width="404"
              height="384"
              fill="none"
              viewBox="0 0 404 384"
              aria-hidden="true"
            >
              <defs>
                <pattern
                  id="64e643ad-2176-4f86-b3d7-f2c5da3b6a6d"
                  x="0"
                  y="0"
                  width="20"
                  height="20"
                  patternUnits="userSpaceOnUse"
                >
                  <rect x="0" y="0" width="4" height="4" class="text-teal-500" fill="currentColor" />
                </pattern>
              </defs>
              <rect width="404" height="384" fill="url(#64e643ad-2176-4f86-b3d7-f2c5da3b6a6d)" />
            </svg>
          </div>
          <div
            class="relative max-w-md mx-auto py-12 px-4 space-y-6 sm:max-w-3xl sm:py-16 sm:px-10 lg:max-w-none lg:p-0 lg:col-start-4 lg:col-span-6"
          >
            <h2 id="join-heading" class="text-4xl font-extrabold text-white">Email Verification</h2>
            <p class="text-xl text-white">A verification email has been sent to you. Please check your inbox!</p>
            <a
              :class="`${isProcessing ? 'pointer-events-none' : ''}`"
              class="block w-full py-3 px-5 text-center bg-white border border-transparent rounded-md shadow-md text-base font-medium text-teal-700 hover:bg-gray-50 sm:inline-block sm:w-auto"
              href="javascript:;"
              data-testid="Send Verification Email Link"
              @click="sendVerificationEmailLink"
              >Resend a Verification Email</a
            >
          </div>
        </div>
      </div>
    </div>
  </div>
</template>
