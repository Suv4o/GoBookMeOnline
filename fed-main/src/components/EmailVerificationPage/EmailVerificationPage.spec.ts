import { render, cleanup, fireEvent } from '@testing-library/vue'
import { shallowMount } from '@vue/test-utils'
import { describe, it, expect, vi } from 'vitest'
import { pinia } from '../../main'
import { useNotification } from '../../utils/composables/notification'
import EmailVerificationPage from './HeroSection.vue'
import { NotificationTypes, useNotificationStore } from '../../store/notification'
import { TestUtils } from '../../utils/helpers'

describe('EmailVerificationPage', async () => {
  it('render component correctly and send verification email link', async () => {
    const useStoreNotification = useNotificationStore(pinia)

    const wrapper = render(EmailVerificationPage, {
      global: {
        plugins: [pinia],
      },
    })

    const buttonSendVerificationEmailLink = wrapper.getByTestId('Send Verification Email Link')

    // Mock sendVerificationEmailLink() function
    TestUtils.Vitest.mockImplementationClick(buttonSendVerificationEmailLink, () => undefined)
    await fireEvent.click(buttonSendVerificationEmailLink)

    expect(buttonSendVerificationEmailLink).toBeTruthy()
    expect(buttonSendVerificationEmailLink).respondTo('click')

    useNotification({
      type: NotificationTypes.Success,
      title: 'Verification Email Sent',
      message: 'A new verification email has been sent to your email address.',
    })

    expect(useStoreNotification.isOpen).toBe(true)
    expect(useStoreNotification.type).toBe(NotificationTypes.Success)
    expect(useStoreNotification.title).toBe('Verification Email Sent')
    expect(useStoreNotification.message).toBe('A new verification email has been sent to your email address.')

    cleanup()
  })

  it('call sendVerificationEmailLink() function', async () => {
    const wrapper = shallowMount(EmailVerificationPage, {
      global: {
        plugins: [pinia],
      },
    })

    const spySendVerificationEmailLink = vi.spyOn(wrapper.vm, 'sendVerificationEmailLink')
    spySendVerificationEmailLink.mockImplementation(() => undefined)
    await wrapper.vm.sendVerificationEmailLink()
    expect(spySendVerificationEmailLink).toHaveBeenCalled()
    expect(spySendVerificationEmailLink).toReturnWith(undefined)
    spySendVerificationEmailLink.mockReset()
    cleanup()
  })

  it('snap shot matches', async () => {
    const wrapper = render(EmailVerificationPage)
    expect(wrapper).toMatchSnapshot()
  })
})
