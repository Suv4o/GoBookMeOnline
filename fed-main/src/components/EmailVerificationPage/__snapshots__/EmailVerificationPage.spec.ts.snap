// Vitest Snapshot v1

exports[`EmailVerificationPage > snap shot matches 1`] = `
{
  "baseElement": <body>
    <div>
      <div
        class="relative pt-28 pb-16 bg-white z-0"
      >
        <div
          aria-hidden="true"
          class="hidden absolute top-0 inset-x-0 h-1/2 bg-gray-50 lg:block"
        />
        <div
          class="max-w-7xl mx-auto bg-teal-600 lg:bg-transparent lg:px-8"
        >
          <div
            class="lg:grid lg:grid-cols-12"
          >
            <div
              class="relative z-10 lg:col-start-1 lg:row-start-1 lg:col-span-4 lg:py-16 lg:bg-transparent"
            >
              <div
                aria-hidden="true"
                class="absolute inset-x-0 h-1/2 bg-gray-50 lg:hidden"
              />
              <div
                class="max-w-md mx-auto px-4 sm:max-w-3xl sm:px-10 lg:max-w-none lg:p-0"
              >
                <div
                  class="aspect-w-10 aspect-h-6 sm:aspect-w-2 sm:aspect-h-1 lg:aspect-w-1"
                >
                  <img
                    alt=""
                    class="object-cover object-center rounded-3xl shadow-2xl"
                    src="/src/assets/images/hero/email-verification.jpg"
                  />
                </div>
              </div>
            </div>
            <div
              class="relative bg-teal-600 lg:col-start-3 lg:row-start-1 lg:col-span-10 lg:rounded-3xl lg:grid lg:grid-cols-10 lg:items-center"
            >
              <div
                aria-hidden="true"
                class="hidden absolute inset-0 overflow-hidden rounded-3xl lg:block"
              >
                <svg
                  aria-hidden="true"
                  class="absolute bottom-full left-full transform translate-y-1/3 -translate-x-2/3 xl:bottom-auto xl:top-0 xl:translate-y-0"
                  fill="none"
                  height="384"
                  viewBox="0 0 404 384"
                  width="404"
                >
                  <defs>
                    <pattern
                      height="20"
                      id="64e643ad-2176-4f86-b3d7-f2c5da3b6a6d"
                      patternUnits="userSpaceOnUse"
                      width="20"
                      x="0"
                      y="0"
                    >
                      <rect
                        class="text-teal-800"
                        fill="currentColor"
                        height="4"
                        width="4"
                        x="0"
                        y="0"
                      />
                    </pattern>
                  </defs>
                  <rect
                    fill="url(#64e643ad-2176-4f86-b3d7-f2c5da3b6a6d)"
                    height="384"
                    width="404"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  class="absolute top-full transform -translate-y-1/3 -translate-x-1/3 xl:-translate-y-1/2"
                  fill="none"
                  height="384"
                  viewBox="0 0 404 384"
                  width="404"
                >
                  <defs>
                    <pattern
                      height="20"
                      id="64e643ad-2176-4f86-b3d7-f2c5da3b6a6d"
                      patternUnits="userSpaceOnUse"
                      width="20"
                      x="0"
                      y="0"
                    >
                      <rect
                        class="text-teal-500"
                        fill="currentColor"
                        height="4"
                        width="4"
                        x="0"
                        y="0"
                      />
                    </pattern>
                  </defs>
                  <rect
                    fill="url(#64e643ad-2176-4f86-b3d7-f2c5da3b6a6d)"
                    height="384"
                    width="404"
                  />
                </svg>
              </div>
              <div
                class="relative max-w-md mx-auto py-12 px-4 space-y-6 sm:max-w-3xl sm:py-16 sm:px-10 lg:max-w-none lg:p-0 lg:col-start-4 lg:col-span-6"
              >
                <h2
                  class="text-4xl font-extrabold text-white"
                  id="join-heading"
                >
                  Email Verification
                </h2>
                <p
                  class="text-xl text-white"
                >
                  A verification email has been sent to you. Please check your inbox!
                </p>
                <a
                  class="block w-full py-3 px-5 text-center bg-white border border-transparent rounded-md shadow-md text-base font-medium text-teal-700 hover:bg-gray-50 sm:inline-block sm:w-auto"
                  data-testid="Send Verification Email Link"
                  href="javascript:;"
                >
                  Resend a Verification Email
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="relative pt-28 pb-16 bg-white z-0"
    >
      <div
        aria-hidden="true"
        class="hidden absolute top-0 inset-x-0 h-1/2 bg-gray-50 lg:block"
      />
      <div
        class="max-w-7xl mx-auto bg-teal-600 lg:bg-transparent lg:px-8"
      >
        <div
          class="lg:grid lg:grid-cols-12"
        >
          <div
            class="relative z-10 lg:col-start-1 lg:row-start-1 lg:col-span-4 lg:py-16 lg:bg-transparent"
          >
            <div
              aria-hidden="true"
              class="absolute inset-x-0 h-1/2 bg-gray-50 lg:hidden"
            />
            <div
              class="max-w-md mx-auto px-4 sm:max-w-3xl sm:px-10 lg:max-w-none lg:p-0"
            >
              <div
                class="aspect-w-10 aspect-h-6 sm:aspect-w-2 sm:aspect-h-1 lg:aspect-w-1"
              >
                <img
                  alt=""
                  class="object-cover object-center rounded-3xl shadow-2xl"
                  src="/src/assets/images/hero/email-verification.jpg"
                />
              </div>
            </div>
          </div>
          <div
            class="relative bg-teal-600 lg:col-start-3 lg:row-start-1 lg:col-span-10 lg:rounded-3xl lg:grid lg:grid-cols-10 lg:items-center"
          >
            <div
              aria-hidden="true"
              class="hidden absolute inset-0 overflow-hidden rounded-3xl lg:block"
            >
              <svg
                aria-hidden="true"
                class="absolute bottom-full left-full transform translate-y-1/3 -translate-x-2/3 xl:bottom-auto xl:top-0 xl:translate-y-0"
                fill="none"
                height="384"
                viewBox="0 0 404 384"
                width="404"
              >
                <defs>
                  <pattern
                    height="20"
                    id="64e643ad-2176-4f86-b3d7-f2c5da3b6a6d"
                    patternUnits="userSpaceOnUse"
                    width="20"
                    x="0"
                    y="0"
                  >
                    <rect
                      class="text-teal-800"
                      fill="currentColor"
                      height="4"
                      width="4"
                      x="0"
                      y="0"
                    />
                  </pattern>
                </defs>
                <rect
                  fill="url(#64e643ad-2176-4f86-b3d7-f2c5da3b6a6d)"
                  height="384"
                  width="404"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="absolute top-full transform -translate-y-1/3 -translate-x-1/3 xl:-translate-y-1/2"
                fill="none"
                height="384"
                viewBox="0 0 404 384"
                width="404"
              >
                <defs>
                  <pattern
                    height="20"
                    id="64e643ad-2176-4f86-b3d7-f2c5da3b6a6d"
                    patternUnits="userSpaceOnUse"
                    width="20"
                    x="0"
                    y="0"
                  >
                    <rect
                      class="text-teal-500"
                      fill="currentColor"
                      height="4"
                      width="4"
                      x="0"
                      y="0"
                    />
                  </pattern>
                </defs>
                <rect
                  fill="url(#64e643ad-2176-4f86-b3d7-f2c5da3b6a6d)"
                  height="384"
                  width="404"
                />
              </svg>
            </div>
            <div
              class="relative max-w-md mx-auto py-12 px-4 space-y-6 sm:max-w-3xl sm:py-16 sm:px-10 lg:max-w-none lg:p-0 lg:col-start-4 lg:col-span-6"
            >
              <h2
                class="text-4xl font-extrabold text-white"
                id="join-heading"
              >
                Email Verification
              </h2>
              <p
                class="text-xl text-white"
              >
                A verification email has been sent to you. Please check your inbox!
              </p>
              <a
                class="block w-full py-3 px-5 text-center bg-white border border-transparent rounded-md shadow-md text-base font-medium text-teal-700 hover:bg-gray-50 sm:inline-block sm:w-auto"
                data-testid="Send Verification Email Link"
                href="javascript:;"
              >
                Resend a Verification Email
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "emitted": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "html": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
